src/Admin/SiteHealth.php

```diff
@@ -0,0 +1,294 @@
+						esc_html__( 'Learn more about persistent object caching', 'amp' ),
+						/* translators: The accessibility text. */
+						esc_html__( '(opens in a new tab)', 'amp' )
+					),
+				]
+			);
+		}
+	}
+
+	/**
+	 * Gets the test result data for whether the curl_multi_* functions exist.
+	 *
+	 * @return array The test data.
+	 */
+	public function curl_multi_functions() {
+		$undefined_curl_functions = array_filter(
+			[
+				'curl_multi_add_handle',
+				'curl_multi_exec',
+				'curl_multi_init',
+			],
+			static function( $function_name ) {
+				return ! function_exists( $function_name );
+			}
+		);
+
+		$data = [
+			'badge'       => [
+				'label' => esc_html__( 'AMP', 'amp' ),
+				'color' => $undefined_curl_functions ? 'orange' : 'green',
+			],
+			'description' => esc_html__( 'The AMP plugin performs better when these functions are available.', 'amp' ),
+			'actions'     => sprintf(
+				'<p><a href="%1$s" target="_blank" rel="noopener noreferrer">%2$s <span class="screen-reader-text">%3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a></p>',
+				esc_url( 'https://www.php.net/manual/book.curl.php' ),
+				esc_html__( 'Learn more about these functions', 'amp' ),
+				/* translators: The accessibility text. */
+				esc_html__( '(opens in a new tab)', 'amp' )
+			),
+			'test'        => 'curl_multi_functions',
+		];
+
+		if ( $undefined_curl_functions ) {
+			return array_merge(
+				$data,
+				[
+					'status'      => 'recommended',
+					'label'       => esc_html(
+						_n(
+							'A curl_multi_* function is not defined',
+							'Some curl_multi_* functions are not defined',
+							count( $undefined_curl_functions ),
+							'amp'
+						)
+					),
+					'description' => wp_kses(
+						sprintf(
+							/* translators: %s: the name(s) of the curl_multi_* PHP function(s) */
+							_n(
+								'The following curl_multi_* function is not defined: %s. The AMP plugin performs better when this function is available.',
+								'The following curl_multi_* functions are not defined: %s. The AMP plugin performs better when these functions are available.',
+								count( $undefined_curl_functions ),
+								'amp'
+							),
+							implode(
+								', ',
+								array_map(
+									static function( $function_name ) {
+										return sprintf( '<code>%s()</code>', $function_name );
+									},
+									$undefined_curl_functions
+								)
+							)
+						),
+						[ 'code' => [] ]
+					),
+				]
+			);
+		} else {
+			return array_merge(
+				$data,
+				[
+					'status' => 'good',
+					'label'  => esc_html__( 'The curl_multi_* functions are defined.', 'amp' ),
+				]
+			);
+		}
+	}
+
+	/**
+	 * Adds debug information for AMP.
+	 *
+	 * @param array $debugging_information The debugging information from Core.
+	 * @return array The debugging information, with added information for AMP.
+	 */
+	public function add_debug_information( $debugging_information ) {
+		add_theme_support( 'amp' );
```

Comment:
Why is `add_theme_support( 'amp' )` here?

Reply:
Hm, maybe I forgot to delete that.

Reply:
It's removed in https://github.com/ampproject/amp-wp/pull/3886/commits/b943f7ec10b824a70c000377bfdc0361ebb78601